frontend/src/metabase/components/SaveQuestionForm/context.tsx

```diff
@@ -62,14 +65,34 @@ export const SaveQuestionProvider = ({
   );
 
   const initialValues: FormValues = useMemo(
-    () => getInitialValues(originalQuestion, question, defaultCollectionId),
-    [originalQuestion, defaultCollectionId, question],
+    () =>
+      getInitialValues(
+        originalQuestion,
+        question,
+        initialCollectionId ?? defaultCollectionId,
```

Comment:
If `initialCollectionId` is passed in, we use that as the initial collection to save to. This mainly applies when the collection picker is visible.